### ISSUE TITLE
conform's automatic type coercion doesn't work with `z.discriminatedUnion`

### ISSUE THREAD

**Issue by jmaldon1** [edmundhung/conform#339]
### Describe the bug and the expected behavior

`z.discriminatedUnion` should be doing automatic type coercion like a normal schema that just uses a `z.object` but it doesn't seem to do that. Everything works fine if i add `.coerce` to the failing fields in my schema.

### Conform version

0.9.0

### Steps to Reproduce the Bug or Issue

```typescript
const schema = z.discriminatedUnion("type", [
  z.object({
    type: z.literal("copy"),
    id: z.number(),
  }),
]);
```

Fails validation with the following error:

```
{
  intent: 'submit',
  payload: { type: 'copy', id: '43' },
  error: { id: [ 'Expected number, received string' ] }
}
```

Form:

```jsx
<Form {...formProps} method="POST">
  <input
    type="hidden"
    name="type"
    value="copy"
  />
  <input
    type="hidden"
    name="id"
    value={id}
  />
  <Button type="submit">Copy</Button>
</Form>
```

**Comment by edmundhung** [comment]
This should be resolved with #340. Thanks for reporting!

**Comment by edmundhung** [comment]
This should be fixed in v1.0.0.
